libs/sdk-core/src/reverseswap.rs

```diff
@@ -239,10 +258,23 @@ impl BTCSendSwap {
 
         match lockup_addr.address_type() {
             Some(AddressType::P2wsh) => {
+                // We explicitly only get the confirmed onchain transactions
+                //
+                // Otherwise, if we had gotten all txs, we risk a race condition when we try
+                // to re-broadcast the claim tx. On re-broadcast, the claim tx is already in the
+                // mempool, so it would be returned in the list below. This however would mark
+                // the utxos as spent, so this address would have a confirmed amount of 0. When
+                // building the claim tx below and trying to subtract fees from the confirmed amount,
+                // this would lead to creating a tx with a negative amount. This doesn't happen
+                // if we restrict this to confirmed txs, because then the mempool claim tx is not returned.
+                //
+                // If the claim tx is confirmed, we would not try to re-broadcast it, so the race
+                // condition only exists when a re-broadcast is tried and the claim tx is unconfirmed.
                 let txs = self
                     .chain_service
-                    .address_transactions(lockup_addr.to_string())
+                    .address_transactions_chain(lockup_addr.to_string())
```

Comment:
I do understand we want to only filter the confirmed transactions but I don't understand why it didn't work with the previous call to `address_transaction` method.
This indeed returns all transactions but bellow we anyway filter only confirmed utxos.

Reply:
In 99% of cases, the version before worked too. The claim tx was usually mined in the very next block, and the user typically didnt trigger `sync()` until then.

However what cause trouble was the the following specific scenario:
- the claim tx was broadcast, but not yet mined
- the SDK received a new `Synced` event

This will then trigger
- monitored reverse swaps (this one) to be handled
- based on the current state, the claim tx will be re-broadcasted
- as part of that, this method (`create_claim_tx`) will be called
- using `chain_service.address_transactions()`[^1] will return all txs of this address, onchain (the confirmed inbound lock tx) and mempool (the pending outbound claim tx)
- the `get_utxos` method on L278 will return an empty list, because the tx output from the lock tx is considered spent by the outgoing claim tx from the mempool
- this leads to `confirmed_amount` below to be 0
- this leads `tx.output[0].value = confirmed_amount - fees;` to panic because it tries to assign a negative number to an `u64` variable

[^1]: Switching to  `chain_service.address_transactions_onchain()` avoids all this, because it ignores the claim tx in the mempool

Reply:
I see now, indeed a slippery bug...
So the logic is right. I would consider though not adding another method (`address_transaction_chain`) as we can use the only `address_transactions` method and filter the confirmed transactions (using the status field) before calculating the utxos.

Reply:
Good idea, I removed the new method and re-used `address_transactions()` with extra filtering.